docs/cpu-target.md

```diff
@@ -41,25 +42,47 @@ To make it possible for slang to produce CPU code, we now need a mechanism to co
+SLANG_SHARED_LIBRARY       ///< A shared library/Dll (for hosting CPU/OS)
+SLANG_HOST_CALLABLE        ///< A CPU target that makes the compiled code available to be run immediately
 ```
 
-These can also be specified on the slang command line as `-target exe` and `-target dll` or `-target sharedlib`.
+These can also be specified on the slang command line as `-target exe` and `-target dll` or `-target sharedlib`. `-target callable` or `-target host-callable` is also possible, but is typically not very useful from the command line, other than to test such code is avaiable for host execution.
 
 In order to be able to use the slang code on CPU, there needs to be binding via values passed to a function that the C/C++ code will produce and export. How this works is described in the ABI section. 
 
 If a binary target is requested, the binary contents will be returned in a ISlangBlob just like for other targets. To use the CPU binary typically it must be saved as file and then potentially marked for execution by the OS before executing. It may be possible to load shared libraries or dlls from memory - but is a non standard feature, that requires unusual work arounds. 
 
 Under the covers when slang is used to generate a binary via a C/C++ compiler, it must do so through the file system. Currently this means that the source (say generated by slang) and the binary (produced by the C/C++ compiler) must all be files. To make this work slang uses temporary files. That the reasoning for hiding this mechanism - and not return say filenames, is so that in the future when binaries are produced directly (for example with LLVM), nothing will need to change.  
 
+Executing CPU Code
+==================
+
+In typically slang operation when code is compiled it produces either source or a binary that can then be loaded by another API such as a rendering API. With CPU code the binary produced could be saved to a file and then executed as an exe or a shared library/dll. In practice though it is not uncommon to want to be able to execute compiled code immediately. Having to save off to a file and then load again can be awkward. It is also not necessarily the case that code needs to be saved to a file to be executed. 
+
+To handle being able call code directly, code can be compiled using the SLANG_HOST_CALLABLE code target type. To access the code that has been produced use the function
+
+```
+    SLANG_API SlangResult spGetEntryPointHostCallable(
+        SlangCompileRequest*    request,
+        int                     entryPointIndex,
+        int                     targetIndex,
+        ISlangSharedLibrary**   outSharedLibrary);
+```        
+
+This outputs a `ISlangSharedLibrary` which whilst in scope, any contained functions remain available (even if the request or session go out of scope). The contained functions can then be accessed via the `findFuncByName` method on the `ISlangSharedLibrary` interface.
```

Comment:
I understand the concern. 

I'm not too worried about it for now for a few reasons

* ISharedLibrary interface is a reasonable abstraction as it stands
* That it is applicable for pass through for C/C++ when there is no meaningful entry point index/name as far as slang is concerned
* That there are mechanisms to get the entry point names through reflection - so I could fairly trivially access (or have a helper on the request on being given the ISlangSharedLibrary and the index gave the function pointer)

That it is true such functionality could be pushed onto the 'thing returned' - but in practice it would mean it would either have to hold a copy of that information or refer to something that did which is a bit clunky.

If the access for an application outlived the request - then the app can just copy the entry point names into its own structures.

Reply:
The one big gotcha I'd point out there is that your current approach is surfacing the implementation decision that CPU entry points will have a dll/so symbol name that is exactly that of the original Slang function (as if declared with `extern "C"`). That is a detail that seems likely to change sooner or later, so that we generate entry point symbols using their *mangled* names, to avoid clashes between modules. I guess if we changed that we'd just add a "get entry point mangled name" function and start telling people to use that, since we'd need the same functionality for DXIL and SPIR-V, so maybe it isn't an argument against the current interface, but it helps explain why I don't like requiring the name string at this point.